src/lib/transform.js

```diff
@@ -62,14 +62,20 @@ function transformRelatedModel(field, prop) {
   return transformer;
 }
 
-function transformGrantModel(field) {
+function transformGrantModel(prop) {
   async function transformer(instance) {
     const obj = {};
-    const activityRecipients = await instance.activityRecipients;
-    if (activityRecipients) {
-      const distinctPS = [...new Set(activityRecipients.map((recipient) => (recipient.grant && recipient.grant[field] !== null ? recipient.grant[field] : '')))];
-      const programSpecialistNames = distinctPS.sort().join('\n');
-      Object.defineProperty(obj, field, {
+    const values = await instance.activityRecipients;
+    if (values) {
+      const distinctList = [
+        ...new Set(
+          values.filter(
+            (recipient) => recipient.grant && recipient.grant[prop] !== null,
+          ).map((r) => r.grant[prop]).flat(),
+        ),
+      ];
+      const programSpecialistNames = distinctList.sort().join('\n');
```

Comment:
Also minor, but since this method is now generic I think we will want a different name for `programSpecialistNames`

Reply:
I have renamed the vars.